### PR TITLE
Adds wlr_data_device_manager destructor

### LINKED ISSUE #399: wlr_data_device_manager_destroy
Should there be a wlr_data_device_manager_destroy? There currently isn't one defined like in the other globals.

### PR DESCRIPTION
**Pull request by Timidger**
Fixes #399

### CODE REVIEW

File: types/wlr_data_device.c [hunk 1]
```diff
@@ -815,3 +815,11 @@ struct wlr_data_device_manager *wlr_data_device_manager_create(
 
 	return manager;
 }
+
+void wlr_data_device_manager_destroy(
+    struct wlr_data_device_manager *manager) {
```
Comment (SirCmpwn) [inline types/wlr_data_device.c]: These would fit on one line?

**Comment by SirCmpwn** [comment]
You also need to free the data_device_manager.

File: types/wlr_data_device.c (hunk 1 above)
Comment (Timidger) [reply types/wlr_data_device.c]: Ah whoops, renamed it and didn't realize it went back under 80 columns

**Comment by Timidger** [comment]
Fixed missing free

**Comment by Timidger** [comment]
Whoops, unrelated init change in there

**Comment by SirCmpwn** [comment]
Squash?

**Comment by Timidger** [comment]
Will do

**Comment by acrisci** [comment]
👍 looks ok

**Pull request merged**